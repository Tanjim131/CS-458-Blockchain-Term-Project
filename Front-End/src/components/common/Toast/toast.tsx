import React, { FC, useEffect, useState } from "react";
import { Alert } from "antd";
import { connect } from "react-redux";
import { showToast } from "../../../../redux/common/action";

/**
 * props for Toast
 */
interface IToastProps {
  toastMessage: string;
  showToast: (show: boolean) => {};
}

/**
 * component for Toast
 * <Toast toastMessage={string}/>
 * @param props
 */
const Toast: FC<IToastProps> = (props: IToastProps) => {
  const { toastMessage, showToast } = props;
  const [showToastMessage, setShowToastMessage] = useState(true);

  /**
   * function for calling the "showToast" action
   * & state of the toastMesage
   */
  function toastActionFunction() {
    setShowToastMessage(false);
    showToast(false);
  }

  useEffect(() => {
    setTimeout(() => toastActionFunction(), 3000);
  }, []);

  return (
    <>
      {showToastMessage && <Alert
        message={toastMessage}
        type="info"
        showIcon={true}
        closable={true}
      />}
    </>
  );
};

export function mapDispatchToProps(dispatch: any) {
  return {
    showToast: (show: boolean) => dispatch(showToast(show)),
  };
}

export default connect(
  null,
  mapDispatchToProps,
)(Toast);
