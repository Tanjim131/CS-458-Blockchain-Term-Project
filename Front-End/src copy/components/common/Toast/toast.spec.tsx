import React from 'react';
import { shallow } from "enzyme";
import Toast, {mapDispatchToProps} from "./toast";
import { shallowToJson } from "enzyme-to-json";

describe('SuccessModal Component', () => {
    let wrapper: any;
    const mockFunc = jest.fn();

    beforeAll(() => {
        wrapper = shallow(
            <Toast.WrappedComponent
                toastMessage="message"
                showToast ={mockFunc}
            />
        )
    })

    it('Check if component exists or not and capture snapshot', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    })

    it("test the mapDispatchToProps", () => {
        const dispatch = jest.fn();        
        mapDispatchToProps(dispatch).showToast(true);
        expect(wrapper).toBeTruthy();
    });

})