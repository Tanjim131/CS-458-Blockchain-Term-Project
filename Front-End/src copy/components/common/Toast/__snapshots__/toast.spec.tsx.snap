// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessModal Component Check if component exists or not and capture snapshot 1`] = `
<Fragment>
  <Alert
    closable={true}
    message="message"
    showIcon={true}
    type="info"
  />
</Fragment>
`;
